requirements needed for pyvista-plot sphinx directive

### Describe the bug, what's wrong, and what you expected.

`pyvista-plot` directive worked with `pip install pyvista` in pyvista<0.40.  In pyvista 0.40, trame package is imported which causes the build to fail.  Using `pip install pyvista[trame]` results in:

```txt
Warning, treated as error:
/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/pyransame/surface.py:docstring of pyransame.surface.random_surface_points:43:Exception occurred in plotting pyransame-random_surface_points-1
 from /home/runner/work/pyransame/pyransame/doc/_stubs/pyransame.random_surface_points.rst:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/pyvista/trame/jupyter.py", line 352, in elegantly_launch
    import nest_asyncio
ModuleNotFoundError: No module named 'nest_asyncio'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/pyvista/ext/plot_directive.py", line 325, in _run_code
    exec(code, ns)
  File "<string>", line 17, in <module>
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/pyvista/plotting/plotter.py", line 6701, in show
    self.last_vtksz = self.export_vtksz(filename=None)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/pyvista/plotting/plotter.py", line 569, in export_vtksz
    elegantly_launch(pyvista.global_theme.trame.jupyter_server_name)
  File "/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/pyvista/trame/jupyter.py", line 354, in elegantly_launch
    raise ImportError(
ImportError: Please install `nest_asyncio` to automagically launch the trame server without await. Or, to avoid `nest_asynctio` run:

    from pyvista.trame.jupyter import launch_server
    await launch_server().ready
make: *** [Makefile:20: html] Error 2
make: Leaving directory '/home/runner/work/pyransame/pyransame/doc'
Error: Process completed with exit code 2.
```

See https://github.com/MatthewFlamm/pyransame/actions/runs/5446695403/jobs/9907797229#step:7:54

### Steps to reproduce the bug.

See https://github.com/MatthewFlamm/pyransame/pull/74

### System Information

```shell
Need to update the CI, but will get.
```


### Screenshots

_No response_

--- a/pyvista/plotting/plotter.py
+++ b/pyvista/plotting/plotter.py
@@ -325,6 +325,7 @@ class BasePlotter(PickingHelper, WidgetHelper):
         self._image_depth_null = None
         self.last_image_depth = None
         self.last_image = None
+        self.last_vtksz = None
         self._has_background_layer = False
         if image_scale is None:
             image_scale = self._theme.image_scale
@@ -6698,7 +6699,10 @@ class Plotter(BasePlotter):
             # always save screenshots for sphinx_gallery
             self.last_image = self.screenshot(screenshot, return_img=True)
             self.last_image_depth = self.get_image_depth()
-            self.last_vtksz = self.export_vtksz(filename=None)
+            try:
+                self.last_vtksz = self.export_vtksz(filename=None)
+            except ImportError:
+                pass
 
         # See: https://github.com/pyvista/pyvista/issues/186#issuecomment-550993270
         if interactive and not self.off_screen:

--- a/pyvista/plotting/utilities/sphinx_gallery.py
+++ b/pyvista/plotting/utilities/sphinx_gallery.py
@@ -123,10 +123,8 @@ class DynamicScraper:
             # else:
             plotter.screenshot(fname)  # produce PNG for thumbnail
             fname = fname[:-3] + "vtksz"
-            if not hasattr(plotter, "last_vtksz"):
-                raise RuntimeError(BUILDING_GALLERY_ERROR_MSG)
             if plotter.last_vtksz is None:
-                raise RuntimeError("Trouble exporting scene, make sure all dependencies are installed.")
+                raise RuntimeError(BUILDING_GALLERY_ERROR_MSG)
             with open(fname, "wb") as f:
                 f.write(plotter.last_vtksz)
             image_names.append(fname)

--- a/pyvista/plotting/utilities/sphinx_gallery.py
+++ b/pyvista/plotting/utilities/sphinx_gallery.py
@@ -6,9 +6,9 @@ import shutil
 import pyvista
 
 BUILDING_GALLERY_ERROR_MSG = (
-    'pyvista.BUILDING_GALLERY must be set to True in your conf.py to capture '
-    'images within sphinx_gallery or when building documentation using the '
-    'pyvista-plot directive.'
+    "pyvista.BUILDING_GALLERY must be set to True in your conf.py to capture "
+    "images within sphinx_gallery or when building documentation using the "
+    "pyvista-plot directive."
 )
 
 
@@ -36,7 +36,7 @@ def html_rst(figure_list, sources_dir, fig_titles='', srcsetpaths=None):
         srcsetpaths = [{0: fl} for fl in figure_list]
 
     figure_paths = [
-        os.path.relpath(figure_path, sources_dir).replace(os.sep, '/').lstrip('/')
+        os.path.relpath(figure_path, sources_dir).replace(os.sep, "/").lstrip("/")
         for figure_path in figure_list
     ]
 
@@ -44,9 +44,9 @@ def html_rst(figure_list, sources_dir, fig_titles='', srcsetpaths=None):
     if len(figure_paths) == 1:
         hinames = srcsetpaths[0]
         srcset = _get_srcset_st(sources_dir, hinames)
-        images_rst = "\n.. offlineviewer:: {}\n\n".format(srcset.lstrip('/'))
+        images_rst = "\n.. offlineviewer:: {}\n\n".format(srcset.lstrip("/"))
     elif len(figure_paths) > 1:
-        raise RuntimeError('Only one figure per output is supported for now.')
+        raise RuntimeError("Only one figure per output is supported for now.")
 
     return images_rst
 
@@ -80,7 +80,7 @@ class Scraper:
         figures = pyvista.plotting.plotter._ALL_PLOTTERS
         for _, plotter in figures.items():
             fname = next(image_path_iterator)
-            if hasattr(plotter, '_gif_filename'):
+            if hasattr(plotter, "_gif_filename"):
                 # move gif to fname
                 fname = fname[:-3] + "gif"
                 shutil.move(plotter._gif_filename, fname)
@@ -125,7 +125,9 @@ class DynamicScraper:
             fname = fname[:-3] + "vtksz"
             if not hasattr(plotter, "last_vtksz"):
                 raise RuntimeError(BUILDING_GALLERY_ERROR_MSG)
-            with open(fname, 'wb') as f:
+            if plotter.last_vtksz is None:
+                raise RuntimeError("Trouble exporting scene, make sure all dependencies are installed.")
+            with open(fname, "wb") as f:
                 f.write(plotter.last_vtksz)
             image_names.append(fname)
         pyvista.close_all()  # close and clear all plotters

